test/checks/lists/listitem.js

```diff
@@ -1,74 +1,81 @@
 describe('listitem', function() {
   'use strict';
 
-  var fixture = document.getElementById('fixture');
   var shadowSupport = axe.testUtils.shadowSupport;
   var checkContext = axe.testUtils.MockCheckContext();
+  var checkSetup = axe.testUtils.checkSetup;
+  var fixtureSetup = axe.testUtils.fixtureSetup;
+  var checkEvaluate = axe.testUtils.getCheckEvaluate('listitem');
 
   afterEach(function() {
-    fixture.innerHTML = '';
     checkContext.reset();
   });
 
   it('should pass if the listitem has a parent <ol>', function() {
-    fixture.innerHTML = '<ol><li id="target">My list item</li></ol>';
-    var target = fixture.querySelector('#target');
-    assert.isTrue(checks.listitem.evaluate.call(checkContext, target));
+    var params = checkSetup('<ol><li id="target">My list item</li></ol>');
+    var result = checkEvaluate.apply(checkContext, params)
+    assert.isTrue(result);
   });
 
   it('should pass if the listitem has a parent <ul>', function() {
-    fixture.innerHTML = '<ul><li id="target">My list item</li></ul>';
-    var target = fixture.querySelector('#target');
-    assert.isTrue(checks.listitem.evaluate.call(checkContext, target));
+    var params = checkSetup('<ul><li id="target">My list item</li></ul>');
+    var result = checkEvaluate.apply(checkContext, params)
+    assert.isTrue(result);
   });
 
   it('should pass if the listitem has a parent role=list', function() {
-    fixture.innerHTML =
-      '<div role="list"><li id="target">My list item</li></div>';
-    var target = fixture.querySelector('#target');
-    assert.isTrue(checks.listitem.evaluate.call(checkContext, target));
+    var params = checkSetup(
+      '<div role="list"><li id="target">My list item</li></div>'
+    );
+    var result = checkEvaluate.apply(checkContext, params)
+    assert.isTrue(result);
   });
 
   it('should pass if the listitem has a parent role=none', function() {
-    fixture.innerHTML =
-      '<ul role="none"><li id="target">My list item</li></ul>';
-    var target = fixture.querySelector('#target');
-    assert.isTrue(checks.listitem.evaluate.call(checkContext, target));
+    var params = checkSetup(
+      '<ul role="none"><li id="target">My list item</li></ul>'
+    );
+    var result = checkEvaluate.apply(checkContext, params)
+    assert.isTrue(result);
   });
 
   it('should pass if the listitem has a parent role=presentation', function() {
-    fixture.innerHTML =
-      '<ul role="presentation"><li id="target">My list item</li></ul>';
-    var target = fixture.querySelector('#target');
-    assert.isTrue(checks.listitem.evaluate.call(checkContext, target));
+    var params = checkSetup(
+      '<ul role="presentation"><li id="target">My list item</li></ul>'
+    );
+    var result = checkEvaluate.apply(checkContext, params)
+    assert.isTrue(result);
   });
 
   it('should fail if the listitem has an incorrect parent', function() {
-    fixture.innerHTML = '<div><li id="target">My list item</li></div>';
```

Comment:
this sort of change makes reviewing just the actual changes tedious

Reply:
The check needs virtual nodes. Can't help that.